### ISSUE TITLE
[BUG] Downloading article (not videos)

### ISSUE THREAD

**Issue by OnlineEdge** [nishad/udemy-dl#75]
Please prefix your issue with one of the following: **[BUG]** **[PROPOSAL]** **[QUESTION]**.

**In raising this issue, I confirm the following (please check boxes):**

- [x] I have read and understood the [contributors guide](https://github.com/nishad/udemy-dl/blob/master/.github/CONTRIBUTING.md).
- [x] I have checked that the bug-fix I am reporting can be replicated, or that the feature I am suggesting isn't already present.
- [x] I have checked that the issue I'm posting isn't already reported.
- [x] I have checked that the issue I'm porting isn't already solved and no duplicates exist in [closed issues](https://github.com/nishad/udemy-dl/issues?q=is%3Aissue+is%3Aclosed) and in [opened issues](https://github.com/nishad/udemy-dl/issues)
- [x] I have checked the pull requests tab for existing solutions/implementations to my issue/suggestion.

**My familiarity with the project is as follows (check one):**

- [ ] I have never used the project.
- [ ] I have used the project briefly.
- [x] I have used the project extensively, but have not contributed previously.
- [ ] I am an active contributor to the project.

**If the project maintainer has any additional requirements, they will be listed here.**

- No additional requirements.

---

Have attempted to downloaded 8 courses, all with different files - Below are the last calls before error, mostly relating to attachments.

```
Traceback (most recent call last):
  File "/usr/local/bin/udemy-dl", line 7, in <module>
    main()
  File "/Library/Python/2.7/site-packages/udemy_dl/udemy_dl.py", line 666, in main
    udemy_dl(username, password, link, lecture_start, lecture_end, save_links, safe_file_names, just_list, output_dest)
  File "/Library/Python/2.7/site-packages/udemy_dl/udemy_dl.py", line 561, in udemy_dl
    get_data(directory, filename, data_url, data_type)
  File "/Library/Python/2.7/site-packages/udemy_dl/udemy_dl.py", line 482, in get_data
    save_art.write(output_html)
UnicodeEncodeError: 'ascii' codec can't encode character u'\u2019' in position 432: ordinal not in range(128)
```

```
Traceback (most recent call last):
  File "/usr/local/bin/udemy-dl", line 7, in <module>
    main()
  File "/Library/Python/2.7/site-packages/udemy_dl/udemy_dl.py", line 666, in main
    udemy_dl(username, password, link, lecture_start, lecture_end, save_links, safe_file_names, just_list, output_dest)
  File "/Library/Python/2.7/site-packages/udemy_dl/udemy_dl.py", line 561, in udemy_dl
    get_data(directory, filename, data_url, data_type)
  File "/Library/Python/2.7/site-packages/udemy_dl/udemy_dl.py", line 478, in get_data
    html = session.get(link).json()['view_html']
KeyError: u'view_html'
```

I then ran a --debug and got the following instantly:

```
Traceback (most recent call last):
  File "/usr/local/bin/udemy-dl", line 7, in <module>
    main()
  File "/Library/Python/2.7/site-packages/udemy_dl/udemy_dl.py", line 666, in main
    udemy_dl(username, password, link, lecture_start, lecture_end, save_links, safe_file_names, just_list, output_dest)
  File "/Library/Python/2.7/site-packages/udemy_dl/udemy_dl.py", line 517, in udemy_dl
    course_id = get_course_id(course_link)
  File "/Library/Python/2.7/site-packages/udemy_dl/udemy_dl.py", line 122, in get_course_id
    save_debug.write(response_clean)
UnicodeEncodeError: 'ascii' codec can't encode character u'\xf1' in position 83856: ordinal not in range(128)
```

Some courses that get this error are:
- https://www.udemy.com/primal-learning/
- https://www.udemy.com/10-facebook-marketing-hacks-that-work-in-2016/

**Comment by mukhlesur-millon** [comment]
please test this
`sudo python -m pip install http://github.com/mukhlesur-millon/udemy-dl/zipball/master -U`

**Comment by OnlineEdge** [comment]
testing now @mukhlesur-millon

**Comment by mukhlesur-millon** [comment]
i haven't add old style article. i think udemy will update all article to new style soon

**Comment by mukhlesur-millon** [comment]
@OnlineEdge did it work?

**Comment by OnlineEdge** [comment]
Got a few running and it is successful across 3 completed ones so far @mukhlesur-millon - amazing work as always!

**Comment by mukhlesur-millon** [comment]
@OnlineEdge please change the tiltle `[BUG] Downloading attachment files (not videos)` to `[BUG] Downloading article (not videos)`

**Comment by nishad** [comment]
@OnlineEdge if the issue is resolved with https://github.com/nishad/udemy-dl/commit/1fd2527275a5f8cdfdbaafe0788f34bf6eff742d, please close this thread.

**Comment by nishad** [comment]
Fixed with v0.2.1. Follow steps in https://github.com/nishad/udemy-dl/blob/master/README.md#update to update.
